### PR TITLE
Streamline JSON parsing

### LINKED ISSUE #933: Streamline JSON-RPC response handling
1. Create `isJsonRpcSuccessResponse` analogue to `isJsonRpcErrorResponse`
2. Create throwing `parseJsonRpcErrorResponse` and `parseJsonRpcSuccessResponse`
3. Deprecate `parseJsonRpcError` and `parseJsonRpcResponse`

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #933

### CODE REVIEW

File: packages/iov-jsonrpc/src/parse.spec.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        const response: any = {
+          jsonrpc: "2.0",
+          error: {
+            code: jsonRpcCode.serverError.default,
+            message: "Something bad happened",
+          },
+        };
+        expect(() => parseJsonRpcErrorResponse(response)).toThrowError(/invalid id field/i);
+      }
+    });
```
Comment (willclarktech) [inline packages/iov-jsonrpc/src/parse.spec.ts]: What if it gets a success response?

File: packages/iov-jsonrpc/src/parse.spec.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      // unset
+      {
+        const response: any = {
+          jsonrpc: "2.0",
+          result: 3000,
+        };
+        expect(() => parseJsonRpcSuccessResponse(response)).toThrowError(/invalid id field/i);
+      }
+    });
+  });
```
Comment (willclarktech) [inline packages/iov-jsonrpc/src/parse.spec.ts]: What if it gets an error response?

File: packages/iov-jsonrpc/src/jsonrpcclient.spec.ts [hunk 3]
```diff
@@ -19,13 +19,13 @@ function makeSimpleMessagingConnection(
...
-        if (responseError) {
-          listener.next(responseError);
-        } else {
-          const response = parseJsonRpcResponse(event.data);
-          listener.next(response);
+        let response: JsonRpcResponse;
+        try {
+          response = parseJsonRpcErrorResponse(event.data);
+        } catch (_) {
+          response = parseJsonRpcSuccessResponse(event.data);
```
Comment (willclarktech) [inline packages/iov-jsonrpc/src/jsonrpcclient.spec.ts]: What if this throws?

File: packages/iov-ethereum/src/httpjsonrpcclient.ts [hunk 4]
```diff
@@ -13,12 +18,12 @@ export class HttpJsonRpcClient {
...
     const responseBody = result.data;
 
-    const errorResponse = parseJsonRpcError(responseBody);
-    if (errorResponse) {
-      return errorResponse;
+    let response: JsonRpcResponse;
+    try {
+      response = parseJsonRpcErrorResponse(responseBody);
+    } catch (_) {
+      response = parseJsonRpcSuccessResponse(responseBody);
```
Comment (willclarktech) [inline packages/iov-ethereum/src/httpjsonrpcclient.ts]: What if this throws?

File: packages/iov-core/src/signingservice.spec.ts [hunk 5]
```diff
@@ -73,13 +73,13 @@ function makeSimpleMessagingConnection(
...
-        if (responseError) {
-          listener.next(responseError);
-        } else {
-          const response = parseJsonRpcResponse(event.data);
-          listener.next(response);
+        let response: JsonRpcResponse;
+        try {
+          response = parseJsonRpcErrorResponse(event.data);
+        } catch (_) {
+          response = parseJsonRpcSuccessResponse(event.data);
```
Comment (willclarktech) [inline packages/iov-core/src/signingservice.spec.ts]: What if this throws?

Comment (webmaster128) [reply packages/iov-core/src/signingservice.spec.ts]: And unhandled exception is thrown and execution stops (as before)

**Comment by webmaster128** [comment]
@willclarktech great feedback, two tests added

**Pull request merged**